### ISSUE TITLE
make no_std a feature

### ISSUE THREAD

**Issue by GopherJ** [Plonky3/Plonky3#290]
currently no_std is enforced in each crate which makes it hard to debug things

**Comment by GopherJ** [comment]
no_std requirement can be removed using the following command
```
rg '#!\[no_std\]' -l | xargs -i sed -i  's|#!\[no_std\]||' {}
```

**Comment by dlubarov** [comment]
I agree it's a bit annoying to not have easy access to things like `dbg!`, although the workarounds don't seem too bad in my opinion - temporarily removing `no_std`, or using `panic!` or something like `tracing::event!`.

I see a couple downsides to conditional use of `no_std`,
1. If we unconditionally import things like `alloc::vec`, we get an unused important warning in builds with std; see [this issue](https://github.com/rust-lang/rust/issues/121708) which mentions some workarounds.
2. We'd probably want to test both builds in CI, doubling CI costs, or the one we don't test might get broken now and then (it was an issue in PLonky2).

So I think my preference would be avoiding conditional use of `no_std` for now, although this could change - i.e. (1) might go away if a nice standard solution develops, while (2) might happen anyway if we end up needing conditional std in order to implement certain optional features in the future.
